Discord fix when a channel is not found. (#1480)
Unstructured-IO/unstructured: CHANGELOG.md

* **Fixes a metadata source serialization bug** Problem: In unstructured elements, when loading an elements json file from the disk, the data_source attribute is assumed to be an instance of DataSourceMetadata and the code acts based on that. However the loader did not satisfy the assumption, and loaded it as a dict instead, causing an error. Fix: Added necessary code block to initialize a DataSourceMetadata object, also refactored DataSourceMetadata.from_dict() method to remove redundant code. Importance: Crucial to be able to load elements (which have data_source fields) from json files.

> * **Fixes issue when channel raises `None`** Retrieves the channel only once, moves `jump_url` property setting within the message retrieval context.

Reviewer: should specify that this is wrt Discord connector. should clearly state the Problem, Fix, and Importance (see the line above this)

Reviewer: (see my earlier comment on this)

> 

Reviewer: follow structure of the log above this one:
title should clearly state that this fixes issue where channel isn't found by the Discord source connector.
then state the Problem: describe what is currently broken and any relevant background. Then state the Fix: what changed in order to fix this. Finally include the Importance: how this impacts the user.

should be generated, however the Formula class inherits from Element instead of Text. After this change the element is correctly created with the correct class 
allowing the document to be loaded. Fix: Change parent class for Formula to Text. Importance: Crucial to be able to load documents that contain formulas.
* **Fixes Sphinx errors.** Fixes errors when running Sphinx `make html` and installs library to suppress warnings.

> * **Fixes issue when channel returns `None`** Problem: Getting the `jump_url` out of the discord `channel` when it's non-existent fails. Fix: property `jump_url`is retrieved within the same context as the messages from the channel. Importance: Avoid possible issues when the connector fails to fetch information about the discord channel.

Reviewer: ```suggestion
* **Fixes issue with Discord connector when a channel returns `None`** Problem: Getting the `jump_url` from a nonexistent Discord `channel` fails. Fix: property `jump_url` is now retrieved within the same context as the messages from the channel. Importance: Avoids cascading issues when the connector fails to fetch information about a Discord channel.
```
